build!: Support net7.0 to remove extensions ambiguous with BCL ones

Youssef1313 opened the pull request:
GitHub Issue (If applicable): #
<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one ore more that apply to this PR -->

<!-- - Bugfix -->
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior?
<!-- Please describe the new behavior after your modifications. -->


## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno.Core/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno.Core/blob/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. 
     Please note that breaking changes are likely to be rejected -->


## Other information
<!-- Please provide any additional information if necessary -->

Internal Issue (If applicable):
<!-- Link to relevant internal issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

jeromelaban commented on src/Directory.Build.props:
```diff
@@ -27,10 +27,11 @@
 			<PrivateAssets>all</PrivateAssets>
 			<IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
-		</PackageReference>
+		</PackageReference> -->
```
We can remove safely, this is not useful anymore.

jeromelaban commented on src/Uno.Core.Extensions.Collections/Uno.Core.Extensions.Collections.csproj:
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras" ToolsVersion="15.0">
 	<PropertyGroup>
-		<TargetFrameworks>uap10.0.17763;uap10.0.19041;net46;netstandard2.0;netstandard2.1;net5.0;net6.0</TargetFrameworks>
+		<TargetFrameworks>uap10.0.17763;uap10.0.19041;net46;netstandard2.0;netstandard2.1;net7.0</TargetFrameworks>
```
Let's keep the intermediate versions as well.

Youssef1313 replied:
@jeromelaban Are they actually needed? They will still remain supported through netstandard I think?

jeromelaban replied:
It would be a binary breaking change for apps that transitively use those versions.

Youssef1313 replied:
I think it can only happen if an API in .NET itself broke binary compat for some API we use? So I feel it's quite unlikely, but added net5.0;net6.0 back anyway.

Youssef1313 committed:
build!: Support net7.0 to remove extensions ambiguous with BCL ones

BREAKING CHANGE: Uno.Extensions.ListExtensions.AsReadOnly is now removed on net7.0 and later as it will be ambiguous with the BCL one (System.Collections.Generic.CollectionExtensions.AsReadOnly). Use the method from BCL.

BREAKING CHANGE: Uno.Extensions.EnumerableExtensions.SkipLast is now removed on net7.0 and later and netstandard2.1 as it will be ambiguous with the BCL one (System.Linq.Enumerable.SkipLast). Use the method from BCL.
